dropshot/src/server.rs

```diff
@@ -642,12 +733,34 @@ impl<C: ServerContext> Future for HttpServer<C> {
     }
 }
 
-impl<C: ServerContext> FusedFuture for HttpServer<C> {
+impl FusedFuture for Joinable {
     fn is_terminated(&self) -> bool {
         self.join_handle.is_none()
     }
 }
 
+impl<C: ServerContext, Closer> Future for HttpServer<C, Closer, Joinable>
+where
+    Closer: Unpin,
+{
+    type Output = Result<(), String>;
+
+    fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
+        let server = Pin::into_inner(self);
+        let joiner = Pin::new(&mut server.joiner);
+        joiner.poll(cx)
+    }
+}
+
+impl<C: ServerContext, Closer> FusedFuture for HttpServer<C, Closer, Joinable>
+where
+    Closer: Unpin,
+{
+    fn is_terminated(&self) -> bool {
+        self.joiner.is_terminated()
+    }
+}
+
```

Comment:
I wonder if we should just remove the impl of `Future` on `HttpServer` so that there's only one way to do all this instead of two.  What do you think?

Reply:
In a vacuum, I'd be fine with this, but doing so would make this a backwards-incompatible change in a pretty significant way.